pimd/pim_cmd.c

```diff
@@ -6413,52 +6486,107 @@ static void show_mroute_summary(struct pim_instance *pim, struct vty *vty)
 		}
 	}
 
-	vty_out(vty, "%-20s %d/%d\n", "(*, G)", starg_hw_mroute_cnt,
-		starg_sw_mroute_cnt + starg_hw_mroute_cnt);
-	vty_out(vty, "%-20s %d/%d\n", "(S, G)", sg_hw_mroute_cnt,
-		sg_sw_mroute_cnt + sg_hw_mroute_cnt);
-	vty_out(vty, "------\n");
-	vty_out(vty, "%-20s %d/%d\n", "Total",
-		(starg_hw_mroute_cnt + sg_hw_mroute_cnt),
-		(starg_sw_mroute_cnt +
-		 starg_hw_mroute_cnt +
-		 sg_sw_mroute_cnt +
-		 sg_hw_mroute_cnt));
+	if (!json) {
+		vty_out(vty, "%-20s %d/%d\n", "(*, G)", starg_hw_mroute_cnt,
+			starg_sw_mroute_cnt + starg_hw_mroute_cnt);
+		vty_out(vty, "%-20s %d/%d\n", "(S, G)", sg_hw_mroute_cnt,
+			sg_sw_mroute_cnt + sg_hw_mroute_cnt);
+		vty_out(vty, "------\n");
+		vty_out(vty, "%-20s %d/%d\n", "Total",
+			(starg_hw_mroute_cnt + sg_hw_mroute_cnt),
+			(starg_sw_mroute_cnt + starg_hw_mroute_cnt
+			 + sg_sw_mroute_cnt + sg_hw_mroute_cnt));
```

Comment:
All of these are unsigned, you need to use the appropriate specifiers and not `%d`.

Reply:
Added suggested change.